nxcomp/src/Timestamp.cpp

```diff
@@ -44,34 +44,57 @@
 
 T_timestamp timestamp;
 
-//
-// The following functions all use the ctime
-// static buffer from the C library.
-//
-
-char *strTimestamp(const T_timestamp &ts)
+std::string strTimestamp(const T_timestamp &ts)
 {
-  char *ctime_now = ctime((time_t *) &ts.tv_sec);
+  std::string ret;
+
+  char ctime_now[26] = { };
+  bool err = true;
+
+#if HAVE_CTIME_S
+  errno_t retval = ::ctime_s(ctime_now, 26, static_cast<const time_t*>(&ts.tv_sec));
```

Comment:
sizeof(ctime)_now instead of hardcoded value?

Reply:
Yep, done.